jboss-integration/src/main/java/org/teiid/jboss/rest/RestASMBasedWebArchiveBuilder.java

```diff
@@ -117,7 +119,280 @@ else if (securityType.equalsIgnoreCase("httpbasic")) {
+            mv = cw.visitMethod(ACC_PUBLIC, "<init>", "()V", null, null);
+            mv.visitVarInsn(ALOAD, 0);
+            mv.visitMethodInsn(INVOKESPECIAL,"java/lang/Object", "<init>", "()V");
+            mv.visitInsn(RETURN);
+            mv.visitMaxs(1, 1);
+            mv.visitEnd();
+        }
+        
+        {
+            mv = cw.visitMethod(ACC_PUBLIC, "init", "(Ljavax/servlet/FilterConfig;)V", null, new String[] {"javax/servlet/ServletException"});
+            av0 = mv.visitAnnotation("Ljava/lang/Override;", true);
+            av0.visitEnd();
+            mv.visitCode();
+            mv.visitInsn(RETURN);
+            mv.visitMaxs(1, 1);
+            mv.visitEnd();
+        }
+        
+        {
+            mv = cw.visitMethod(ACC_PUBLIC, "doFilter", "(Ljavax/servlet/ServletRequest;Ljavax/servlet/ServletResponse;Ljavax/servlet/FilterChain;)V", null, new String[] {"javax/servlet/ServletException", "java/io/IOException"});
+            av0 = mv.visitAnnotation("Ljava/lang/Override;", true);
+            av0.visitEnd();
+            mv.visitCode();
+            mv.visitVarInsn(ALOAD, 2);
+            mv.visitTypeInsn(CHECKCAST, "javax/servlet/http/HttpServletResponse");
+            mv.visitVarInsn(ASTORE, 4);
+            mv.visitVarInsn(ALOAD, 4);
+            mv.visitLdcInsn(k1);
+            mv.visitLdcInsn(v1);
+            mv.visitMethodInsn(INVOKEINTERFACE, "javax/servlet/http/HttpServletResponse", "addHeader", "(Ljava/lang/String;Ljava/lang/String;)V");
+            mv.visitVarInsn(ALOAD, 4);
+            mv.visitLdcInsn(k2);
+            mv.visitLdcInsn(v2);
+            mv.visitMethodInsn(INVOKEINTERFACE, "javax/servlet/http/HttpServletResponse", "addHeader", "(Ljava/lang/String;Ljava/lang/String;)V");
+            mv.visitVarInsn(ALOAD, 4);
+            mv.visitLdcInsn(k3);
+            mv.visitLdcInsn(v3);
+            mv.visitMethodInsn(INVOKEINTERFACE, "javax/servlet/http/HttpServletResponse", "addHeader", "(Ljava/lang/String;Ljava/lang/String;)V");
+            mv.visitVarInsn(ALOAD, 3);
+            mv.visitVarInsn(ALOAD, 1);
+            mv.visitVarInsn(ALOAD, 2);
+            mv.visitMethodInsn(INVOKEINTERFACE, "javax/servlet/FilterChain", "doFilter", "(Ljavax/servlet/ServletRequest;Ljavax/servlet/ServletResponse;)V");
+            mv.visitInsn(RETURN);
+            mv.visitMaxs(6, 3);
+            mv.visitEnd();
+        }
+        
+        {
+            mv = cw.visitMethod(ACC_PUBLIC, "destroy", "()V", null, null);
+            av0 = mv.visitAnnotation("Ljava/lang/Override;", true);
+            av0.visitEnd();
+            mv.visitCode();
+            mv.visitInsn(RETURN);
+            mv.visitMaxs(1, 1);
+            mv.visitEnd();
+        }
+        
+        cw.visitEnd();
+        
+        return cw.toByteArray();
+           
+    }
+	
+	protected byte[] getBootstrapServletClass(String vdbName, String desc, String version, String[] schamas, String baseUrl, String packages, Boolean scan) {
```

Comment:
Is there any *dynamic* content in here? if there not any, we need to add this as  .java class, and then use that.